### PR TITLE
feat(openfeature): add LRU cache for exposure deduplication

### PR DESCRIPTION
**Pull request by leoromanovsky**
### Motivation

Right now we're sending an exposure event every time a flag gets evaluated, even if the same user is getting the same variant over and over. This creates a lot of noise - if a user hits the same flag 100 times in a session, we don't need 100 identical exposure events.

The Eppo SDK uses a deduplication cache to solve this. The idea is simple: only send an exposure when something actually changes (new user, different variant, different allocation). This keeps the exposure data meaningful without flooding the backend.

We have some system tests written to exercise this which the golang sdk is current failing: https://github.com/DataDog/system-tests/pull/5868

### Changes

Added an LRU cache to the exposure writer that tracks `(flag, subject) -> (allocation, variant)`:

  - Cache implementation - Simple LRU using container/list with a default capacity of 65536 entries. When we see a flag evaluation, we check if we've already logged this exact combination. If yes, skip it. If it's new or the allocation/variant changed, log it and update the cache. To decided to use `container/list` so we can use a standard library.
  - Thread safety - The cache itself isn't thread-safe, but all access goes through exposureWriter.append() which holds a mutex. This keeps the check-and-update atomic so concurrent evaluations don't cause duplicate exposures.
  - Tests - Added a bunch of unit tests covering the cache behavior: same subject deduplication, different subjects, variant/allocation cycling, LRU eviction, edge cases. Also added concurrent access tests to make sure the locking actually works under load.

The unit tests in this repo matches the deduplication semantics in the system-tests.

Prior implementations:

* https://github.com/Eppo-exp/golang-sdk/blob/main/eppoclient/lruassignmentlogger.go
* datadog java exposure cache - https://github.com/DataDog/dd-trace-java/blob/master/products/feature-flagging/lib/src/main/java/com/datadog/featureflag/ExposureCache.java

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality at or near 100% coverage.
- [ ] [System-Tests](https://github.com/DataDog/system-tests/) covering this feature have been added and enabled with the va.b.c-dev version tag.
- [ ] There is a benchmark for any new code, or changes to existing code.
- [ ] If this interacts with the agent in a new way, a system test has been added.
- [ ] New code is free of linting errors. You can check this by running `./scripts/lint.sh` locally.
- [ ] Add an appropriate team label so this PR gets put in the right place for the release notes.
- [ ] Non-trivial go.mod changes, e.g. adding new modules, are reviewed by @DataDog/dd-trace-go-guild.

Unsure? Have a question? Request a review!

### CODE REVIEW

File: openfeature/exposure.go [hunk 1]
```diff
@@ -89,10 +95,79 @@ type exposurePayload struct {
...
+	capacity int
+	items    map[string]*list.Element
+	order    *list.List // front = most recently used, back = least recently used
+}
+
+// newExposureLRUCache creates a new LRU cache with the given capacity
+func newExposureLRUCache(capacity int) *exposureLRUCache {
+	return &exposureLRUCache{
+		capacity: capacity,
+		items:    make(map[string]*list.Element),
```
Comment (dd-oleksii) [inline openfeature/exposure.go]: minor: Go allows indexing maps with structs, so we can define:
```go
type exposureCacheKey struct {
    flagKey string
    targetingKey string
}
```

and use that to index into items.

Pros:
- less string allocations/formattting
- no chance of accidental collision if flag key contains `|`

File: openfeature/exposure.go [hunk 2]
```diff
@@ -42,6 +43,11 @@ const (
 
 	// defaultHTTPTimeout is the timeout for HTTP requests to the agent
 	defaultHTTPTimeout = 5 * time.Second
+
+	// defaultExposureCacheCapacity is the default capacity for the exposure deduplication cache.
+	// 65536 (2^16) balances memory efficiency (~6.5MB max) with sufficient capacity for
```
Comment (dd-oleksii) [inline openfeature/exposure.go]: nitpicking: I don't think we have limits on flag/variant/subject id size, so max estimate cannot be made

File: openfeature/exposure.go [hunk 3]
```diff
@@ -89,10 +95,79 @@ type exposurePayload struct {
...
+	}
+}
+
+// add adds or updates an entry in the cache and returns true if this is a new
+// or changed entry (should generate exposure), false if it's a duplicate
+func (c *exposureLRUCache) add(key, allocationKey, variant string) bool {
+	if elem, exists := c.items[key]; exists {
+		// Entry exists - check if allocation/variant changed
+		entry := elem.Value.(*exposureCacheEntry)
+		if entry.allocationKey == allocationKey && entry.variant == variant {
```
Comment (dd-oleksii) [inline openfeature/exposure.go]: nitpicking: similarly to `exposureCacheKey`, it would be slightly more maintainable if we define `exposureCacheValue` type.

File: openfeature/exposure.go [hunk 4]
```diff
@@ -89,10 +95,79 @@ type exposurePayload struct {
...
+			// Same allocation and variant - this is a duplicate, move to front
+			c.order.MoveToFront(elem)
```
Comment (dd-oleksii) [inline openfeature/exposure.go]: nitpick: we can call `c.order.MoveToFront(elem)` before the `if`
